src/Microsoft.ML.Dnn/TensorTypeExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using NumSharp.Backends;
+using NumSharp.Backends.Unmanaged;
+using NumSharp.Utilities;
+using Tensorflow;
+
+namespace Microsoft.ML.Transforms
+{
+    [BestFriend]
+    internal static class TensorTypeExtensions
+    {
+        public static void ToScalar<T>(this Tensor tensor, ref T dst) where T : unmanaged
+        {
+            if (typeof(T).as_dtype() == tensor.dtype)
```

Comment:
is it expected to just no-op when the types mis-match? Would it be better to throw an exception?